### PR TITLE
add seissolxdmfwriter pip module

### PR DESCRIPTION
**Pull request by Thomas-Ulrich**
I realize we need to write seissol output is several of our python scripts.
E.g.
https://github.com/SeisSol/SeisSol/pull/353
https://github.com/SeisSol/SeisSol/pull/360
or 
https://github.com/SeisSol/Meshing/blob/master/vizualizeBoundaryConditions/vizualizeBoundaryConditions.py
Therefore I uploaded a small script on pypi for that.

### CODE REVIEW

File: seissolxdmfwriter/LICENSE [hunk 1]
```diff
@@ -0,0 +1,29 @@
+BSD 3-Clause License
+
+Copyright (c) 2020, SeisSol Group
```
Comment (sebwolf-de) [inline seissolxdmfwriter/LICENSE]: Should be 2021 ;-)

**Review by sebwolf-de — COMMENTED** [review]
Hi, thanks for this script and sorry for not reviewing this earlier...
I just have one pedantic comment about the license....

File: seissolxdmfwriter/seissolxdmfwriter/seissolxdmfwriter.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    myDtype = "float64" if precision == "double" else "float32"
+    nCells, node_per_element = connect.shape
+    with h5py.File(prefix + ".h5", "w") as h5f:
+        h5f.create_dataset("mesh0/connect", (nCells, node_per_element), dtype="i8")
+        h5f["mesh0/connect"][:, :] = connect[:, :]
+        h5f.create_dataset("mesh0/geometry", xyz.shape, dtype="d")
+        h5f["mesh0/geometry"][:, :] = xyz[:, :]
+        for k, dataName in enumerate(lDataName):
+            hdname = "mesh0/" + dataName
+            h5f.create_dataset(hdname, (len(lidt), nCells), dtype=myDtype)
```
Comment (sebwolf-de) [inline seissolxdmfwriter/seissolxdmfwriter/seissolxdmfwriter.py]: I just wanted to write integer data. I propose to remove `precision` and add
```
dtypeDict = {"int64": "i8", "int32": "i4", "float64": "float64", "float32": "float32"}
...
    h5f.create_dataset(hdname, (len(lidt), nCells), dtype=dtypeDict[lData[k].dtype.name])
```

File: seissolxdmfwriter/LICENSE (hunk 1 above)
Comment (Thomas-Ulrich) [reply seissolxdmfwriter/LICENSE]: done

File: seissolxdmfwriter/seissolxdmfwriter/seissolxdmfwriter.py (hunk 2 above)
Comment (Thomas-Ulrich) [reply seissolxdmfwriter/seissolxdmfwriter/seissolxdmfwriter.py]: I've integrated your changes

File: seissolxdmfwriter/seissolxdmfwriter/seissolxdmfwriter.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+     <DataItem NumberType="Float" Precision="8" Format="HDF" Dimensions="{nNodes} 3">{prefix}.h5:/geometry</DataItem>
+    </Geometry>
+    <Time Value="{idt*dt}"/>"""
+        for k, dataName in enumerate(lDataName):
+            prec = 4 if reduce_precision else precisionDict[lData[k].dtype.name]
+            number_type = numberTypeDict[lData[k].dtype.name]
+            xdmf += f"""
+    <Attribute Name="{dataName}" Center="Cell">
+     <DataItem ItemType="HyperSlab" Dimensions="{nCells}">
+      <DataItem NumberType="UInt" Precision="4" Format="XML" Dimensions="3 2">{i} 0 1 1 1 {nCells}</DataItem>
```
Comment (sebwolf-de) [inline seissolxdmfwriter/seissolxdmfwriter/seissolxdmfwriter.py:32]: Should be `NumberType="{number_type}" Precision="{prec}"`

Comment (Thomas-Ulrich) [reply seissolxdmfwriter/seissolxdmfwriter/seissolxdmfwriter.py:32]: Here is an extract of a fault output:
``` xml
    <Attribute Name="SRs" Center="Cell">
     <DataItem ItemType="HyperSlab" Dimensions="6256">
      <DataItem NumberType="UInt" Precision="4" Format="XML" Dimensions="3 2">0 0 1 1 1 6256</DataItem>
      <DataItem NumberType="Float" Precision="8" Format="HDF" Dimensions="1 6256">tpv13-fault_cell.h5:/mesh0/SRs</DataItem>
     </DataItem>
    </Attribute>
```
So I think my code is consistent with seissol (while I don't understand why it is like that)

**Review by sebwolf-de — APPROVED** [review]

**Pull request merged**